### PR TITLE
:sparkles: Enhance tag handling on forms

### PR DESCRIPTION
**Pull request by sjd78**
## Summary

Enhance tag handling on forms to:
  - Display the select list options as "Tag Category Name / Tag Name"
  - Sort the options by the display name
  - Add a tooltip to the selected tag labels to show the tag's category
  - Sort the selected tag labels by the display name

Resolves: https://issues.redhat.com/browse/MTA-1533

### Enhance Tag handling on Application form, Tag and Stakeholder handling on Archetype form,  and Stakeholder handling on Assessment Wizard

Tag fields will now:
  - Show the tag category and tag name together in the select list
  - Show the tag category and tag name on the selected label list
  - Provide a tooltip on the tag with the name of the tag's tag category
  - use a complex object as the form value type and default values: `TagItemType`
  - make sure only `Ref` values get put in the assessment payload
  - use a `TagItemType[]` provided by a new query for the list of options to work with

Stakeholders and Stakeholder Groups fields will now:
  - use `Ref` as the form value type and default values
  - make sure only `Ref` values get put in the assessment payload
  - transform full data objects into items specifically formatted for use by `Autocomplete`

This uses refactoring in `Autocomplete` to support keeping a complex id and name object in the form values.

Form value types, default value calculations, and `onValidSubmit` payload building have been updated as needed.  Form values are using complex objects with at least `{ id, name }` data.  Default values can push any objects that match the minimum form value complex object type.  The payload builds make sure that the form values are converted to exactly a `Ref` object to satisfy hub's strict data marshaling.

### Refactor `Autocomplete` and `HookFormAutocomplete`

  - Use a complex object for its `options` and `selections` props.  This allows for a richer experience with the typeahead component, select list, and selected label list.

  - Minimize the states in use and to replace `useEffect` in favor of `useMemo`.

  - An `AutocompleteOptionProp` allows rendering a name, a separate name in the selected label list, and a tooltip for the selected label.

  - Renamed `ItemsSelect` to `HookFormAutocomplete` since all it needs to do now is wrap a react-hook-form around an Autocomplete

### Update application and archetype rest api functions and queries

  - Update api functions `createApplication` and `createArchetype` to take as input a `New<>` payload and return the newly created entity as a result.

  - Update query `useCreateApplicationMutation` to pass the proper values to the `onSuccess` and `onError` callbacks.

  - Update various fetch queries to return `isSuccess` so component code can ensure at least 1 successful fetch has happened before rendering data that depends on the fetch results.

  - Add query `useFetchTagsWithTagItems` as a convenience to get a full list of tags with tag category refs, and a `TagItemType` transform of the tags to be useful for display.

## Screen Shots
**Manual Tags select list on Application Create:**
image

**Manual Tags showing existing selected tags:**
image

**Update Archetype form with existing selections for Criteria tags, Archetype Tags, Stakeholder(s) and Stakeholder Group(s) and a mouse hover on a selected criteria tag showing the tooltip:**
image

### CODE REVIEW

File: client/src/app/components/Autocomplete.tsx [hunk 1]
```diff
@@ -266,12 +266,40 @@ export const Autocomplete: React.FC<IAutocompleteProps> = ({
   const menu = (
     <Menu
       ref={menuRef}
-      onSelect={onSelect}
-      onKeyDown={handleMenuKeyDown}
+      // onSelect={handleMenuOnSelect}
```
Comment (gitdallas) [inline client/src/app/components/Autocomplete.tsx:269]: Seems to be handled at the menuItem now which I understand is why this is commented out. Cleanup/remove?

File: client/src/app/components/Autocomplete.tsx [hunk 2]
```diff
@@ -286,23 +314,35 @@ export const Autocomplete: React.FC<IAutocompleteProps> = ({
...
-                {currentSelection}
-              </Label>
+          {selectedOptions.map(({ id, name, labelName, tooltip }) => (
+            <FlexItem key={id}>
+              <LabelToolip content={tooltip}>
+                <Label
+                  color={labelColor}
+                  onClose={() => deleteSelectionByItemId(id)}
+                >
+                  {labelName ? toString(labelName) : toString(name)}
```
Comment (gitdallas) [inline client/src/app/components/Autocomplete.tsx:329]: just a nit
```suggestion
                  {toString(labelName || name)}
```

**Review by gitdallas — APPROVED** [review]
lgtm couple nits

File: client/src/app/components/Autocomplete.tsx (hunk 1 above)
Comment (sjd78) [reply client/src/app/components/Autocomplete.tsx:269]: I forgot to take it out.  Good eye!

**Pull request merged**